mobile/ios/ios-device.ts

```diff
@@ -380,5 +420,79 @@ export class IOSDevice implements Mobile.IIOSDevice {
 
 		}).future<void>()();
 	}
+
+	public uninstallApplication(applicationId: string): IFuture<void> {
+		return (() => {
+			var afc = this.startService(iOSProxyServices.MobileServices.INSTALLATION_PROXY);
+			try {
+				var result = this.$mobileDevice.deviceUninstallApplication(afc, this.$coreFoundation.createCFString(applicationId), null, this.uninstallApplicationCallbackPtr);
+				if(result !== 0) {
+					this.$errors.fail("AMDeviceUninstallApplication returned '%d'.", result);
+				}
+			} finally {
+			}
+		}).future<void>()();
+	}
+
+	private resolveAfc(): Mobile.IAfcClient {
+		var service = options.app ?  this.startHouseArrestService(options.app) : this.startService(iOSProxyServices.MobileServices.APPLE_FILE_CONNECTION);
+		var afcClient: Mobile.IAfcClient = this.$injector.resolve(iOSProxyServices.AfcClient, {service: service});
+
+		return afcClient;
+	}
+
+	public getFile(deviceFilePath: string): IFuture<void> {
+		return (() => {
+			var afcClient = this.resolveAfc();
+			var fileToRead = afcClient.open(deviceFilePath, "r");
+			var fileToWrite = process.stdout;
+
+			var size = 0;
+			while(true) {
+				var data = fileToRead.read(8192);
+				if(!data || data.length === 0) {
+					break;
+				}
+				fileToWrite.write(data);
+				size += data.length;
+			}
+
+			fileToRead.close();
+			this.$logger.info("%s bytes read from %s", size.toString(), deviceFilePath);
+
+		}).future<void>()();
+	}
+
+	public putFile(localFilePath: string, deviceFilePath: string): IFuture<void> {
+		var afcClient = this.resolveAfc();
+		return afcClient.transfer(localFilePath, deviceFilePath);
+	}
+
+	public listFiles(devicePath: string): IFuture<void> {
+		return (() => {
+			if(!devicePath) {
+				devicePath = ".";
+			}
+
+			this.$logger.info("Listing %s", devicePath);
+
+			var afcClient = this.resolveAfc();
+
+			var walk = (root: string, indent: number) => {
+				this.$logger.info(util.format("%s %s", Array(indent).join(" "), root));
+				var children: any[] = [];
```

Comment:
The function is declared to return string[], do we need to declare the var as any[]?